File: programmers/lv0/코딩기초트레이닝/조건문/이어붙인수/Solution.js
/**
 * @param {int[]}num_list
 * @returns {number}
 */
const solution = (num_list) => {
    const {evens, odds} = num_list.reduce((acc, curr) => {
        if (curr % 2 === 0) acc.evens.push(curr)
        else acc.odds.push(curr)
        return acc
    }, {
        evens: [],
        odds: [],
    })
    return Number(evens.join('')) + Number(odds.join(''))
}

console.log(solution([3, 4, 5, 2, 1]))
console.log(solution([5, 7, 8, 3]))

File: programmers/lv0/코딩기초트레이닝/조건문/이어붙인수/Solution.swift
import Foundation

func solution(_ numList: [Int]) -> Int {
    let evens = numList.filter { $0 % 2 == 0 }.map { String($0) }.joined()
    let odds = numList.filter { $0 % 2 != 0 }.map { String($0) }.joined()

    return (Int(evens) ?? 0) + (Int(odds) ?? 0)
}